beacon-chain/types/state.go

```diff
@@ -33,13 +34,19 @@ func NewActiveState(data *pb.ActiveState) *ActiveState {
 }
 
 // NewGenesisStates initializes a beacon chain with starting parameters.
-func NewGenesisStates() (*ActiveState, *CrystallizedState) {
+func NewGenesisStates() (*ActiveState, *CrystallizedState, error) {
+	// Bootstrap recent block hashes to all 0s for first 2 cycles (128 slots).
+	var recentBlockHashes [][]byte
+	for i := 0; i < 2*params.CycleLength; i++ {
+		recentBlockHashes = append(recentBlockHashes, make([]byte, 0, 32))
+	}
 	active := &ActiveState{
 		data: &pb.ActiveState{
 			PendingAttestations: []*pb.AttestationRecord{},
-			RecentBlockHashes:   [][]byte{},
+			RecentBlockHashes:   recentBlockHashes,
```

Comment:
This seems wasteful. Couldn't validation skip `recentBlockHashes` indexes if `block.slot_number` < `params.CycleLength * 2` instead of allocating a bunch of empty byte arrays? This would mean that the first 128 blocks all have empty arrays.